lib/proxy.rb

```diff
@@ -58,7 +58,12 @@ def process_token_or_authenticate!(env)
     if token = request.params['token']
       content_id = process_token(token, env)
     end
-    authenticate!(env) unless content_id
+
+    if env['warden']
```

Comment:
I get the impression this is only here because `env['warden']` hasn't been set for the tests. (i.e. in production it will always exist). I'll see if I can set it up.

Reply:
Ah sure, some prior art is https://github.com/alphagov/publishing-api/blob/master/spec/support/authentication_helper.rb#L4-L8 which is wired in via: https://github.com/alphagov/publishing-api/blob/master/spec/rails_helper.rb#L65